app/Http/Controllers/RESTController.php

```diff
@@ -0,0 +1,290 @@
+    {
+        return $request->route()[2];
+    }
+
+    /**
+     * Execute the usecase that the controller prepared.
+     *
+     * @throws HTTP_Exception_400
+     * @throws HTTP_Exception_403
+     * @throws HTTP_Exception_404
+     * @return void
+     */
+    protected function executeUsecase()
+    {
+        try {
+            // Attempt to execute the usecase to get the response
+            $responsePayload = $this->usecase->interact();
+
+            return $responsePayload;
+        } catch (\Ushahidi\Core\Exception\NotFoundException $e) {
+            abort(404, $e->getMessage());
+        } catch (\Ushahidi\Core\Exception\AuthorizerException $e) {
+            abort(403, $e->getMessage());
+        } catch (\Ushahidi\Core\Exception\ValidatorException $e) {
+            throw new ValidationException($e->getMessage(), $e);
+        } catch (\InvalidArgumentException $e) {
+            abort(400, "Bad request: ". $e->getMessage());
+        }
+    }
+
+    /**
+     * Prepare response headers and body, formatted based on user request.
+     * @throws HTTP_Exception_400
+     * @throws HTTP_Exception_500
+     * @return void
+     */
+    protected function prepResponse(Array $responsePayload = null, Request $request)
+    {
+        // Add CORS headers to the response
+        // $this->add_cors_headers($this->response);
+
+        // Use JSON if the request method is OPTIONS
+        if ($request->method() === Request::METHOD_OPTIONS) {
+            $type = 'json';
+        } else {
+            //...Get the requested response format, use JSON for default
+            // @todo should just Accept header
+            $type = strtolower($request->query('format')) ?: 'json';
+        }
+
+        try {
+            //$format = service("formatter.output.$type");
+
+            // $body = $format($this->_response_payload);
+            // $mime = $format->getMimeType();
+            // $this->response->headers('Content-Type', $mime);
+
+            if (empty($responsePayload)) {
+                // If the payload is empty, return a 204
+                // https://tools.ietf.org/html/rfc7231#section-6.3.5
+                $response = response('', 204);
+            } else {
+                $response = response()->json(
+                    $responsePayload,
+                    200,
+                    [],
+                    env('APP_DEBUG', false) ? JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES : null
+                );
+
+                if ($type === 'jsonp') {
+                    $response->withCallback($request->input('callback'));
+                    // Prevent Opera and Chrome from executing the response as anything
+                    // other than JSONP, see T455.
+                    $response->headers->set('X-Content-Type-Options', 'nosniff');
+                }
+            }
+
+            // Should we prevent this request from being cached?
+            if (! in_array($request->method(), $this->cacheableMethods)) {
```

Comment:
This is straight from the Kohana version, but in short no. The web server knows nothing about the content its serving, so it doesn't know if its safe to cache or not.
I think originally this was used to make sure GET responses were cachable, but POST/PUT/DELETE were not. I suspect its not actually working now